R/add_keep_population_flag.R

```diff
@@ -0,0 +1,163 @@
+#' Add keep_popluation flag
+#'
+#' @description Add keep_population flag to individual files
+#' @param data A data frame
+#'
+#' @return A data frame with keep_population flags
+#' @family individual_file
+
+add_keep_population_flag <- function(individual_file, year) {
+  calendar_year <- paste0("20", substr(year, 1, 2)) %>% as.integer()
+
+  if (!check_year_valid(year, "NSU")) {
+    individual_file <- individual_file %>%
+      dplyr::mutate(keep_population = 1L)
+  } else {
+    ## Obtain the population estimates for Locality AgeGroup and Gender.
+    pop_estimates <-
+      readRDS(get_datazone_pop_path("DataZone2011_pop_est_2011_2021.rds")) %>%
+      dplyr::as_tibble() %>%
+      dplyr::select(year, datazone2011, sex, age0:age90plus)
+
+    # Step 1: Obtain the population estimates for Locality, AgeGroup, and Gender
+    # Select out the estimates for the year of interest.
+    # if we don't have estimates for this year (and so have to use previous year).
+    year_available <- pop_estimates %>%
+      dplyr::pull(year) %>%
+      unique()
+    if (calendar_year %in% year_available) {
+      pop_estimates <- pop_estimates %>%
+        dplyr::filter(year == calendar_year)
+    } else {
+      previous_year <- sort(year_available, decreasing = TRUE)[1]
+      pop_estimates <- pop_estimates %>%
+        dplyr::filter(year = previous_year)
+    }
+
+    pop_estimates <- pop_estimates %>%
+      # Recode gender to make it match source.
+      dplyr::mutate(sex = dplyr::if_else(sex == "M", 1, 2)) %>%
+      dplyr::rename(
+        "age90" = "age90plus",
+        "gender" = "sex"
+      ) %>%
+      tidyr::pivot_longer(
+        names_to = "age",
+        names_prefix = "age",
+        values_to = "population_estimate",
+        cols = "age0":"age90"
+      ) %>%
+      dplyr::mutate(
+        age = as.integer(age),
+        age_group = dplyr::case_when(
+          age >= 0 & age <= 4 ~ "0-4",
+          age >= 5 & age <= 14 ~ "5-14",
+          age >= 15 & age <= 24 ~ "15-24",
+          age >= 25 & age <= 34 ~ "25-34",
+          age >= 35 & age <= 44 ~ "35-44",
+          age >= 45 & age <= 54 ~ "45-54",
+          age >= 55 & age <= 64 ~ "55-64",
+          age >= 65 & age <= 74 ~ "65-74",
+          age >= 75 & age <= 84 ~ "75-84",
+          age >= 85 ~ "85+"
+        )
+      ) %>%
+      dplyr::left_join(
+        get_locality_path() %>%
+          readRDS() %>%
```

Comment:
I think this could be re-written as..
```suggestion
        readr::read_rds(get_locality_path()) 
```